src/ParticleContainer.h

```diff
@@ -0,0 +1,116 @@
+//
+// Created by Celik Yucel on 23.10.24.
+//
+#pragma once
+
+#include "Particle.h"
+
+#include <vector>
+
+/**
+ * Container for storing Particle objects and
+ * to provide iterators.
+ */
+class ParticleContainer {
+
+    private:
+    /**
+    * Vector for storing particles
+    */
+    std::vector<Particle> particles;
+
+    /**
+    * Vector for storing particle pairs
+    */
+    std::vector<std::pair<std::reference_wrapper<Particle>, std::reference_wrapper<Particle>>> particlePairs;
+
+    public:
+    /////////////////////////////////////////////(Iterators)///////////////////////////////////////////////////////////
```

Comment:
is this a standard practice? does it create a nice view in IDEs? 
If not, then I don't see much point to it.